src/dune_rules/odoc.ml

```diff
@@ -314,6 +319,12 @@ let setup_library_odoc_rules cctx (local_lib : Lib.Local.t) =
   let ctx = Super_context.context sctx in
   let* requires = Compilation_context.requires_compile cctx in
   let info = Lib.Local.info local_lib in
+  let modes = Lib_info.modes info in
+  let mode =
```

Comment:
this needs to be done the other way around:

- it's only `` `Melange `` if `modes.melange == true` and the OCaml modes are false.

Reply:
we have `Lib_mode.Map.Set.for_merlin`. Looks like we want to share the same logic.

Reply:
I ended up just re-using `Lib_mode.Map.Set.for_merlin`. Is that fine ?

Reply:
Yup! that's even better. I wonder if we should rename it now that it's not for merlin only.